File: Mathematical/Geeks4Geeks/Find the median/find_the_median.py
class Solution:
	def find_median(self, v):
	    size = len(v)
	    v.sort()
	    if size % 2 == 0:
	        middle1 = v[size // 2]
	        middle2 = v[size // 2 - 1]
	        return (middle1 + middle2) // 2
	    else:     
	        return v[size // 2]

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		n = int(input())
		v = list(map(int,input().split())) 
		ob = Solution();
		ans = ob.find_median(v)
		print(ans)

File: Mathematical/Geeks4Geeks/Find the median/find_the_median.java
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine().trim());
        while(T-->0)
        {
            int n = Integer.parseInt(br.readLine().trim());
            String s = br.readLine();
            String[] S = s.split(" ");
            int[] v  = new int[n];
            for(int  i = 0; i < n; i++)
                v[i] = Integer.parseInt(S[i]);
            Solution ob = new Solution();
            int ans = ob.find_fact(v);
            System.out.println(ans);
        }
    }
}

class Solution
{
    public int find_fact(int[] v)
    {
        Arrays.sort(v);
        int n = v.length;
        if(n%2 != 0) return (int)(v[n/2]);
        else return (int)(v[(n-1)/2] + v[n/2]) / 2;
    }
}
